backend/src/main/java/com/htecgroup/skynest/service/impl/PermissionServiceImpl.java

```diff
@@ -228,13 +233,42 @@ public void revokePermission(UUID bucketId, String userEmail) {
     checkIfBucketIsDeleted(bucketId);
     currentUserHasPermissionForBucket(bucketId, AccessType.OWNER);
 
-    UserEntity user = userRepository.findUserByEmail(userEmail).orElseThrow(UserNotFoundException::new);
+    UserEntity user =
+        userRepository.findUserByEmail(userEmail).orElseThrow(UserNotFoundException::new);
+    checkIfTargetUserIsTheCurrentUser(user);
+    UserObjectAccessEntity permission =
+        permissionRepository
+            .findByObjectIdAndGrantedTo(bucketId, user)
+            .orElseThrow(BucketNotFoundException::new);
+    checkIfPermissionExist(permission);
+    permissionRepository.delete(permission);
+  }
+
+  @Override
+  public void revokeFilePermission(UUID fileId, String email) {
+    FileMetadataEntity fileMetadataEntity =
+        fileRepository.findById(fileId).orElseThrow(FileNotFoundException::new);
+    checkIfBucketIsDeleted(fileMetadataEntity.getBucket().getId());
+    checkIfFileIsDeleted(fileMetadataEntity);
```

Comment:
if the bucket is deleted, this would return true
I don't think you need `checkIfBucketIsDeleted`